scripts/zkai_native_seq32_attention_mlp_dry_run_opening_sampler_gate.py

```
@@ -0,0 +1,672 @@
+            }
+        )
+    return output.getvalue()
+
+
+def ensure_output_path(path: pathlib.Path) -> None:
+    target = pathlib.Path(os.path.abspath(path))
+    evidence = EVIDENCE_DIR.resolve()
+    try:
+        target.relative_to(evidence)
+    except ValueError as err:
+        raise DryRunOpeningSamplerGateError("output path must stay under evidence directory") from err
```

Ruff (0.15.13)

[warning] Avoid specifying long messages outside the exception class

(TRY003)